event-infinite-scroll.html

```diff
@@ -16,114 +22,132 @@
+         * @event reach-top
+         */
+
+        properties: {
             /**
-             * Fired when reach bottom.
+             * track scroll on ready
+             * otherwise you can call observer when ever you want
              *
-             * @event reach-bottom
+             * @boolean autoStart
              */
+            autoStart: {
+                type: Boolean,
+                value: false,
+                observer: '_startScroll'
+            },
 
             /**
-             * Fired when reach top.
+             * The offset is used for triggering the notifications
+             * before the real top/bottom is reached
              *
-             * @event reach-top
+             * @int scrollOffset
              */
-
-            properties: {
-                /**
-                 * track scroll on ready
-                 * otherwise you can call startObserve when ever you want
-                 *
-                 * @boolean readyOnStart
-                 */
-                readyOnStart: {
-                    type: Boolean,
-                    value: true
-                },
-
-                /**
-                 * When you want to notification
-                 *
-                 * @int scrollDenominator
-                 */
-                scrollDenominator: {
-                    type: Number,
-                    value: 1
-                },
-
-                /**
-                 * You can set your item loading delay time when scroll up infinite
-                 *
-                 * @int loadingDelay
-                 */
-                loadingDelay: {
-                    type: Number,
-                    value: 1
-                }
+            scrollOffset: {
+                type: Number,
+                value: 200,
+                reflectToAttribute: true
             },
 
-            observerStatus: true,
-            observer: null,
+            /**
+             * You can set your item loading delay time when scroll infinite
+             *
+             * @int loadingDelay (in milliseconds)
+             */
+            loadingDelay: {
+                type: Number,
+                value: 0,
+                reflectToAttribute: true
+            }
+        },
+
+        observerStatus: true,
+        observer: null,
 
-            loadingDelay: 1,
+        /**
+         * Triggered when the autoStart status is changed
+         *
+         * @readyOnStart [boolean] autoStart
+         */
+        _startScroll: function( newValue ) {
+            if ( newValue ) {
+                this.startObserve();
+            }
+        },
+
+        startObserve: function () {
+            this.parentNode.addEventListener( "scroll", function () {
+                this.scrolled();
+            }.bind( this ) );
+
+            this.parentNode.addEventListener( "mousewheel", function () {
+                this.scrolledWheel();
+            }.bind( this ) );
+
+            var self = this;
```

Comment:
what is this significance of `var self = this;` if you bind `this` and use `this` within the closure anyway?